l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+        });
+        Diamond.DiamondCutData memory diamondCut = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: address(0),
+            initCalldata: hex""
+        });
+        bytes memory bytecode = abi.encodePacked(
+            type(DiamondProxy).creationCode,
+            abi.encode(config.chainId, diamondCut)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("DiamondProxy deployed at:", contractAddress);
+        addresses.stateTransition.diamondProxy = contractAddress;
+    }
+
+    function deployErc20BridgeProxy() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(addresses.bridgehub.bridgehubProxy, addresses.transparentProxyAdmin, bytes(hex""))
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("Erc20BridgeProxy deployed at:", contractAddress);
+        addresses.bridges.erc20BridgeProxy = contractAddress;
+    }
+
+    function deploySharedBridgeContracts() internal {
+        deploySharedBridgeImplementation();
+        deploySharedBridgeProxy();
+        registerSharedBridge();
+    }
+
+    function deploySharedBridgeImplementation() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(L1SharedBridge).creationCode,
+            // solhint-disable-next-line func-named-parameters
+            abi.encode(
+                config.tokens.tokenWethAddress,
+                addresses.bridgehub.bridgehubProxy,
+                addresses.bridges.erc20BridgeProxy,
+                config.eraChainId,
+                addresses.bridges.erc20BridgeImplementation,
+                addresses.stateTransition.diamondProxy
+            )
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("SharedBridgeImplementation deployed at:", contractAddress);
+        addresses.bridges.sharedBridgeImplementation = contractAddress;
+    }
+
+    function deploySharedBridgeProxy() internal {
+        uint256 storageSwitch = config.contracts.sharedBridgeUpgradeStorageSwitch;
+        bytes memory initCalldata = abi.encodeCall(L1SharedBridge.initialize, (addresses.governance, storageSwitch));
+        bytes memory bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(addresses.bridges.sharedBridgeImplementation, addresses.transparentProxyAdmin, initCalldata)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("SharedBridgeProxy deployed at:", contractAddress);
+        addresses.bridges.sharedBridgeProxy = contractAddress;
+    }
+
+    function registerSharedBridge() internal {
+        Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.startBroadcast();
+        bridgehub.addToken(address(0x01));
```

Comment:
there is a constant for this address at the top of the file

Reply:
The constant is in DeployHyperchain script. Added it anyway to this script as well for consistency.